plugin/plugin_settings.py

```diff
@@ -282,9 +282,9 @@ def populate_include_dirs(self, view):
         # replace project related variables to real ones
         for i, include_dir in enumerate(include_dirs):
             include_dir = re.sub(
-                "(\$project_base_path)", self.project_base_folder, include_dir)
-            include_dir = re.sub("(\$project_name)",
-                                 self.project_base_name, include_dir)
+                r"\$project_base_path", re.escape(self.project_base_folder), include_dir)
+            include_dir = re.sub(r"\$project_name",
+                                 re.escape(self.project_base_name), include_dir)
```

Comment:
This is a good intention but execution is probably not exactly correct. Using re.escape for the string replace argument is probably not correct because you don't use that string for RE matching. re.escape should only be used for making string work in a regexp pattern. This is a replacement, not a pattern.

It probably would be enough to just do self.project_base_name.replace(r'\', r'\\')